### ISSUE TITLE
'Hosts' tests does not take into account 'cluster' scope

### ISSUE THREAD

**Issue by equelin** [WahlNetwork/Vester#50]
In each `hosts` tests (like Update-DNS) we do not look at the cluster scope  

``` Powershell
foreach ($server in (Get-VMHost -Name $config.scope.host)) {
...
}
```

Instead, I propose do to something like:

``` Powershell
foreach ($server in (Get-Cluster -Name $config.scope.cluster | Get-VMHost -Name $config.scope.host)) {
...
}
```

What do you think about that ?

**Comment by tvanholland** [comment]
I would go a step further and add an additional scope of 'Datacenter'.

`foreach ($server in (Get-Datacenter -name $config.scope.datacenter | Get-Cluster -Name $config.scope.cluster | Get-VMHost -Name $config.scope.host)) {
...
}`

**Comment by brianbunke** [comment]
I'll do you guys one better (worse?). Assume this scenario:

1. `.scope.host = *`
2. `.scope.cluster = *`
3. `.vcenter.vc = 'vCenterDev'`
4. `$global:DefaultVIServers = @(vCenterDev, vCenterProd)`

Because we aren't adding the -Server parameter to Get-VMHost, Get-Cluster, etc., we're now searching (and remediating?) against both vCenters.

Travis is also right...we probably should add in a `.scope.datacenter` in case someone doesn't want to wildcard at that level.

I want to fix this, but I've been thinking of it as just one of a few action items to address when the tests get refactored for the module. If anyone wants to hit it now, that's great too.

**Comment by equelin** [comment]
> I want to fix this, but I've been thinking of it as just one of a few action items to address when the tests get refactored for the module. If anyone wants to hit it now, that's great too.

Don't you think we should have a todo list somewhere to gather all changes/functionnalities we want to add to the module ?

**Comment by brianbunke** [comment]
Yeah. Not sure the best way to do that. I'd rather not make it a markdown file that we have to pull against to update. Maybe that Projects tab? (Haven't used it yet, so I don't know, but it seems relevant.)

**Comment by chriswahl** [comment]
I think a tracking list that references any issues by their number is a great idea. Would help organize sprints (or at least versions and releases) moving forward.

**Comment by chriswahl** [comment]
On topic: My original thought was that the user can specify the cluster and whatnot within the variables.

`.scope.host = Get-Cluster @("Alpha","Bravo") | Get-VMHost`

However, I never described that anywhere. Perhaps because my environment is so simple I didn't think too much about it. I'd be fine with creating a hierarchy of variables with vcenter, data center, cluster, and host information if we think that'd be easier to define.

**Comment by tvanholland** [comment]
The VM hierarchy would probably be the most complex.
`foreach ($VM in (Get-Datacenter $config.scope.datacenter | Get-Cluster $config.scope.cluster | Get-VM $config.scope.vm -server $config.vc.vcenter))`

If this looks good i'll make the change for the rest of them.

**Comment by equelin** [comment]
Sounds good to me except that I will use the `-server` parameter to the `Get-Datacenter` command. I feel that it'll be more efficient to directly select the right datacenter.

**Comment by equelin** [comment]
If everyone is ok, This issue should be closed due to PR #63

**Comment by chriswahl** [comment]
👍
